manim/mobject/geometry.py

```diff
@@ -812,3 +859,233 @@ class RoundedRectangle(Rectangle):
     def __init__(self, **kwargs):
         Rectangle.__init__(self, **kwargs)
         self.round_corners(self.corner_radius)
+
+
+class Tiling(VMobject):
+    """
+    The purpose of this class is to create tilings/tesselations.
+    Tilings can also be seemlessly transformed into each other.
+    This requires their ranges to be the same, tiles to be oriented and
+    rotated the correct way, as well as having a vertex setup that
+    allows proper transformations.
+    
+    Parameters
+    ----------
+    tile_prototype : Mobject or function(x,y) that returns a Mobject
+    x_offset : nested list of Mobject methods and values
```

Comment:
well this is kinda annoying to give a type to but...
i feel like if we used tuples indicating a function and its values, it'd be easier to deal with; might be easier to use and more self-evident as well

without changing anything:
```suggestion
    x_offset : Iterable[Iterable[Union[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
```
with tuples:
```suggestion
    x_offset : Iterable[Iterable[Tuple[Callable[[:class:`int`, :class:`int`], :class:`~.Mobject`]]]]
```